### ISSUE TITLE
Memory timeline should report metrics for the relevant browser only

### ISSUE THREAD

**Issue by perezju** [catapult-project/catapult#2026]
The memory_timeline metric should only report results for memory dumps that include the renderer process which requested the dump. This becomes relevant in scenarios where multiple browsers may be active during a single benchmark.

**Comment by perezju** [comment]
@primiano @nedn @eakuefner

**Comment by nedn** [comment]
@natduca I think this could be another usage of "manual interaction record".

**Comment by perezju** [comment]
Yes. To fix this my CL https://codereview.chromium.org/1712633002/ relies on the fact that TimelineBasedMetric.AddResults is called only with the interaction records relevant to a particular  renderer_thread. And my benchmark https://codereview.chromium.org/1711713002/ tags each dump with a different interaction record.

**Comment by natduca** [comment]
Hmmm I think a design doc here would be good, and taking our time to socialize the concern more would be very good.

**Comment by perezju** [comment]
I started this issue only to track a small change needed on the existing memory-infra (on the Telemetry side) for a new experimental benchmark to work. The wider picture to consider, I guess, is how multi-browser support would work in the new world of TBMv2. At the moment is probably premature to write a design doc for that, but I started already some email thread to bounce ideas about it, and will be glad to write the design doc itself after we gain a bit more experience with TBMv2 metrics.
